mousestyles/path_diversity/smooth_noise.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import print_function, absolute_import, division
+
+import pandas as pd
+import numpy as np
+from path_features import compute_angles
+from detect_noise import detect_noise
+
+
+def smooth_noise(movement, paths, angle_threshold, delta_t):
+    r"""
+    Return a new movement pandas DataFrame object containing
+    CT, CX, CY coordinates and the homebase status. The points at which
+    noise, as defined by the input parameters, is detected have been
+    smoothed by averaging.
+
+    Parameters
+    ----------
+    movement : pandas.DataFrame
+        CT, CX, CY coordinates and homebase status
+        for the unique combination of strain, mouse and day
+
+    paths index : a list containing the indices for all paths
+
+    angle_threshold : float
+        positive number indicating the minimum turning angle to flag as noise
+
+    delta_t : float
+        positive number indicating the delta_time interval
+
+    Returns
+    -------
+    smoothed movement : pandas.DataFrame
+        CT, CX, CY coordinates and homebase status
+        for the unique combination of strain, mouse and day
+
+    Examples
+    --------
+    >>> movement = data.load_movement(1, 2, 1)
+    >>> paths = path_index(movement, 1, 1)
+    >>> smoothed_movement = smooth_noise(movement, paths, 135, .1)
+    """
+
+    # check if all inputs are positive
+    conditions_value = [angle_threshold < 0, delta_t < 0]
+    if any(conditions_value):
+        raise ValueError("Input values need to be positive")
+
+    if not isinstance(movement, pd.core.frame.DataFrame):
+        raise TypeError("Movement must be pandas DataFrame")
+
+    if set(movement.keys()) != {'isHB', 't', 'x', 'y'}:
```

Comment:
This could be a problem when you pass the `movement` object which also has the column for food or water. I don't know whether or not we are going to use those additional columns, but I would like to suggest to use
`if set(movement.keys()).issuperset(['x','y','t']) == False:`
which works similarly but is more robust for existence of those additional columns.